dataprofiler/tests/profilers/test_unstructured_labeler_profile.py

```diff
@@ -219,7 +220,70 @@ def test_unstructured_labeler_profile_add(self, mock):
+    @mock.patch('dataprofiler.profilers.'
+                'unstructured_labeler_profile.DataLabeler')
+    @mock.patch('dataprofiler.profilers.'
+                'unstructured_labeler_profile.'
+                'CharPostprocessor')
+    def test_diff(self, mock1, mock2):
+        """
+        Tests to see that entity percentages match the counts given
+        """
+        profiler1 = UnstructuredLabelerProfile()
+        profiler1.char_sample_size = 20
+        profiler1.word_sample_size = 10
+        profiler1.entity_counts["postprocess_char_level"]["UNKNOWN"] = 6
+        profiler1.entity_counts["postprocess_char_level"]["TEST"] = 14
+        profiler1.entity_counts["true_char_level"]["UNKNOWN"] = 4
+        profiler1.entity_counts["true_char_level"]["TEST"] = 16
+        profiler1.entity_counts["word_level"]["UNKNOWN"] = 5
+        profiler1.entity_counts["word_level"]["TEST"] = 5
+        profiler1.update(pd.Series(["a"]))
+        profile1 = profiler1.profile
+        entity_counts1 = profile1['entity_counts']
+        entity_percentages1 = profile1['entity_percentages']
+
+        profiler2 = UnstructuredLabelerProfile()
+        profiler2.char_sample_size = 10
+        profiler2.word_sample_size = 5
+        profiler2.entity_counts["postprocess_char_level"]["UNKNOWN"] = 6
+        profiler2.entity_counts["postprocess_char_level"]["TEST"] = 4
+        profiler2.entity_counts["true_char_level"]["UNKNOWN"] = 8
+        profiler2.entity_counts["true_char_level"]["TEST"] = 2
+        profiler2.entity_counts["word_level"]["UNKNOWN"] = 1
+        profiler2.entity_counts["word_level"]["TEST"] = 4
+        profile2 = profiler2.profile
+        entity_counts2 = profile2['entity_counts']
+        entity_percentages2 = profile2['entity_percentages']
+
+        expected_diff = {
+            'entity_counts' : {
+                'postprocess_char_level' : utils.find_diff_of_dicts(
+                            entity_counts1['postprocess_char_level'],
+                            entity_counts2['postprocess_char_level']),
+                'true_char_level': utils.find_diff_of_dicts(
+                    entity_counts1['true_char_level'],
+                    entity_counts2['true_char_level']),
+                'word_level': utils.find_diff_of_dicts(
+                    entity_counts1['word_level'],
+                    entity_counts2['word_level'])
+            },
+            'entity_percentages': {
+                'postprocess_char_level' : utils.find_diff_of_dicts(
+                    entity_percentages1['postprocess_char_level'],
+                    entity_percentages2['postprocess_char_level']),
+                'true_char_level': utils.find_diff_of_dicts(
+                    entity_percentages1['true_char_level'],
+                    entity_percentages2['true_char_level']),
+                'word_level': utils.find_diff_of_dicts(
+                    entity_percentages1['word_level'],
+                    entity_percentages2['word_level'])
```

Comment:
We should manually set these as I'm not sure using the function in the test which is within the function is appropriate.

Reply:
I thought this was the method we decided on since we have some floating point precision errors and assertDictEquals fails because on that. The past diff PRs use the util functions in the tests as well, thoughts?

Reply:
Shouldn't floating point errors not be a concern though since the values you chose would make predictable, calculatable percentages (6/20 = 0.3, 14/20 = 0.7, ...)?

Reply:
This was discussed last time, but 0.7 - 0.4 became 0.2999..., which failed checks. Typing that into the Python console results in that as well.

Reply:
This is a bit of an ask but it may be worth it in the future, we could write our own util function for dict equality that uses almost equal on floats? Cuz this seems like it could be a recurring problem down the line that would be helpful for other tests too. @JGSweets thoughts?